apps/dashboard/src/main/java/com/akto/action/JiraIntegrationAction.java

```diff
@@ -376,6 +361,231 @@ private BasicDBObject buildContentDetails(String txt, String link) {
         return details;
     }
 
+    String aktoDashboardHost;
+    List<TestingIssuesId> issuesIds;
+    private String errorMessage;
+    public String bulkCreateJiraTickets (){
+        if(issuesIds == null || issuesIds.isEmpty()){
+            addActionError("Cannot create an empty jira issue.");
+            return ERROR.toUpperCase();
+        }
+
+        if((projId == null || projId.isEmpty()) || (issueType == null || issueType.isEmpty())){
+            addActionError("Project ID or Issue Type cannot be empty.");
+            return ERROR.toUpperCase();
+        }
+
+        JiraIntegration jiraInstance = JiraIntegrationDao.instance.findOne(new BasicDBObject());
+        if(jiraInstance == null) {
+            addActionError("Jira is not integrated.");
+            return ERROR.toUpperCase();
+        }
+
+        List<JiraMetaData> jiraMetaDataList = new ArrayList<>();
+        Bson projection = Projections.include(YamlTemplate.INFO);
+        List<String> testingSubCategories = new ArrayList<>();
+        for(TestingIssuesId testingIssuesId : issuesIds) {
+            testingSubCategories.add(testingIssuesId.getTestSubCategory());
+        }
+        List<YamlTemplate> yamlTemplateList = YamlTemplateDao.instance.findAll(Filters.in("_id", testingSubCategories), projection);
+        Map<String, Info> testSubTypeToInfoMap = new HashMap<>();
+        for(YamlTemplate yamlTemplate : yamlTemplateList) {
+            Info info = yamlTemplate.getInfo();
+            testSubTypeToInfoMap.put(info.getSubCategory(), info);
+        }
+
+        List<TestingRunResult> testingRunResultList = new ArrayList<>();
+        int existingIssues = 0;
+        for(TestingIssuesId testingIssuesId : issuesIds) {
+            TestingRunIssues issue = TestingRunIssuesDao.instance.findOne(Filters.and(
+                    Filters.in("_id", testingIssuesId),
```

Comment:
find all issues first, in a single query with projections, dont put load on db that much, and try reducing heap space as well